pytorch_lightning/loops/fit_loop.py

```diff
@@ -0,0 +1,254 @@
+
+    def reset(self) -> None:
+        self.iteration_count = 0
+
+    def run(self):
+        if not self._should_skip_training():
+            return super().run()
+
+    def on_run_start(self):
+        self.results.to(device=self.trainer.lightning_module.device)
+        self.trainer.call_hook("on_train_start")
+
+    def on_advance_start(self):
+        model = self.trainer.lightning_module
+
+        # reset train dataloader
+        if self.current_epoch != 0 and self.trainer.reload_dataloaders_every_epoch:
+            self.trainer.reset_train_dataloader(model)
+
+        # TODO: specify the possible exception
+        with suppress(Exception):
+            # set seed for distributed sampler (enables shuffling for each epoch)
+            self.trainer.train_dataloader.sampler.set_epoch(self.current_epoch)
+
+        # changing gradient according accumulation_scheduler
+        self.trainer.accumulation_scheduler.on_train_epoch_start(self.trainer, self.trainer.lightning_module)
+
+        # stores accumulated grad fractions per batch
+        self.training_loop.batch_loop.accumulated_loss = TensorRunningAccum(
+            window_length=self.trainer.accumulate_grad_batches
+        )
+
+        # hook
+        self.trainer.logger_connector.on_epoch_start()
+        self.trainer.call_hook("on_epoch_start")
+        self.trainer.call_hook("on_train_epoch_start")
+
+    def advance(self):
+        train_dataloader = self.trainer.accelerator.process_dataloader(self.trainer.train_dataloader)
+        train_dataloader = self.trainer.data_connector.get_profiled_train_dataloader(train_dataloader)
+
+        with self.trainer.profiler.profile("run_training_epoch"):
+            # run train epoch
+            epoch_output = self.training_loop.run(train_dataloader)
+            # log epoch metrics
+
+            if epoch_output is None:
+                return
+
+            # the global step is manually decreased here due to backwards compatibility with existing loggers
+            # as they expect that the same step is used when logging epoch end metrics even when the batch loop has
+            # finished. this means the attribute does not exactly track the number of optimizer steps applied.
+            # TODO(@carmocca): deprecate and rename so users don't get confused
+            self.global_step -= 1
+            # log epoch metrics
+            self.trainer.logger_connector.update_train_epoch_metrics()
+            self.global_step += 1
+
+    def on_advance_end(self):
+        if self.training_loop.batches_seen == 0:
+            return
+
+        self.training_loop.update_lr_schedulers('epoch')
+
+        did_train_only = self.trainer.disable_validation or self.trainer.evaluation_loop.should_skip_evaluation(
+            self.trainer.num_val_batches
+        )
+        if did_train_only:
+            self.global_step -= 1
+            self.check_checkpoint_callback(True)
+            self.global_step += 1
```

Comment:
keep the TODO about removing this whole check for `did_train_only` ?

Reply:
I can't find a todo in the old training loop where this code is. Was something lost?

Reply:
ah it was below in the check_checkpoint_callback, nvm!